File: server/src/main/java/com/springboot/store/entity/PurchaseOrder.java
package com.springboot.store.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder

@Entity
@Table(name = "purchase_order")
public class PurchaseOrder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private int subtotal;
    private int discount;
    private int total;
    private String note;
    private Date createdDate;

    @OneToMany(mappedBy = "purchaseOrder", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<PurchaseOrderDetail> purchaseOrderDetail;

    @ManyToOne
    @JoinColumn(name = "staff_id")
    private Staff staff;

    @ManyToOne
    @JoinColumn(name = "supplier_id")
    private Supplier supplier;

    @ManyToOne
    @JoinColumn(name = "store_id")
    private Store store;
}

File: entities/PurchaseOrder.ts
export type PurchaseOrder = {
  id: number;
  subtotal: number;
  discount: number;
  total: number;
  note: string;
  createdDate: string;
  purchaseOrderDetail: PurchaseOrderDetail[];
  staffId: number;
  supplierId: number;
};

export type PurchaseOrderDetail = {
  id: number;
  quantity: number;
  price: number;
  discount: number;
  productId: number;
};
